### DISCUSSION TITLE
Best way to express markush representation?

### DISCUSSION THREAD

**Discussion by orgw** [rdkit/rdkit#6934] [Q&A]
Hi i wanted to ask what is the best way to express a markush representation? The purpose it to make one and filter molecules that match the representation. for instance something like this:

![image](https://github.com/rdkit/rdkit/assets/64889801/216758d4-bd5b-4778-a672-50311c0fa9d2)

maybe GenericGroups.h can be a start?

any ideas will be a great help

thanks

**Comment by DavidACosgrove** [comment]
Hi @orgw,
Encoding the Markush structures from patents, as this appears to be, is non-trivial, to say the least.  For example, if restrictions are imposed on the combinations of U, V and W that can occur, you will have to enumerate them individually.  If Z is a variable length piece, that will also potentially create an enumeration problem.  The other features might be amenable to wildcards/generic groups in this case.  If you want to see how knotty it can get, see https://pubs.acs.org/doi/10.1021/ci3000387.  The system described in this paper is not freely available (it is owned by AstraZeneca), sadly.
Dave

**Reply by orgw** [reply]
Hi thanks for the reference paper. I agree that this issue isn't easy. I was wondering if there were any attempts in using tree-based representations in approaching this problem?
